packages/number-field/src/vaadin-number-field.js

```diff
@@ -377,42 +357,33 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
     return !this.value || (!this.disabled && this._incrementIsInsideTheLimits(incr, value));
   }
 
-  /** @private */
-  __applyStep(step) {
-    if (this.inputElement) {
-      this.inputElement.step = this.__validateByStep ? step : 'any';
-    }
-  }
-
   /**
-   * @param {number} newVal
-   * @param {number | undefined} oldVal
+   * @param {number} step
    * @protected
    */
-  _stepChanged(newVal) {
-    // TODO: refactor to not have initial value
-    // https://github.com/vaadin/vaadin-text-field/issues/435
-
-    // Avoid using initial value in validation
-    this.__validateByStep = this.__stepChangedCalled || this.getAttribute('step') !== null;
-
-    this.__applyStep(newVal);
+  _stepChanged(step, inputElement) {
+    if (typeof step === 'string') {
+      const stepNumber = parseFloat(step);
+      this.step = isNaN(stepNumber) ? null : stepNumber;
+      return;
+    }
```

Comment:
Previously, the type casting was achieved by setting the `step` attribute. This part is most likely not relevant anymore but `integer-field` has a special test that fails if I drop the type-casting:

https://github.com/vaadin/web-components/blob/48e5e39670645eba5b9d0f9be91e29b134b93bf8/packages/integer-field/test/integer-field.test.js#L196-L199

Thinking rationally, I don't see why `step` should be a special case in terms of type-casting: why it should allow strings for `step` while they are not allowed for `min` and `max`. It seems to be more of the user's responsibility not to set something inappropriate given that the property has the type `Number`. Otherwise, pretty much every property can be a candidate for having such type-casting.

So I believe we could drop this logic along with the failing `integer-field` test. Any other considerations?

Reply:
Current logic indeed inconsistent: we have tests for accepting `value` and `step` as strings, but we don't have such tests and logic for `min` and `max`. There are no comments regarding this in the PR: https://github.com/vaadin/vaadin-text-field/pull/415/

Let's change the logic to only accept `step` as a number, and update Flow counterparts accordingly if needed.

Reply:
Removed.